### PR TITLE
Commentate Interpreter (no-op)

### PR DESCRIPTION
**Pull request by StachuDotNet**

### CODE REVIEW

File: fsharp-backend/src/LibExecution/Interpreter.fs [hunk 1]
```diff
@@ -33,10 +35,12 @@ let withGlobals (state : ExecutionState) (symtable : Symtable) : Symtable =
 let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
   // Design doc for execution results and previews:
   // https://www.notion.so/darklang/Live-Value-Branching-44ee705af61e416abed90917e34da48e
-  // TODO document is either gone or hidden behind login
+  // TODO remove link from code or avail document - it is either gone or hidden behind login
   let sourceID id = SourceID(state.tlid, id)
   let incomplete id = DIncomplete(SourceID(state.tlid, id))
 
+  /// WHATISTHIS?
+  /// "if we're previewing some execution, ...?"
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibExecution/Interpreter.fs:43]: I'm having a hard time understanding the purpose of, and describing, this function. Could you help? @pbiggar 
Full fn:
https://github.com/darklang/dark/blob/662e99b8cb70a48b1019503d8dba772f323b0f07/fsharp-backend/src/LibExecution/Interpreter.fs#L44-L50

File: fsharp-backend/src/LibExecution/Interpreter.fs [hunk 2]
```diff
@@ -72,10 +76,11 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
       // We ignore incompletes but not error rail.
       // CLEANUP: Other places where lists are created propagate incompletes
       // instead of ignoring, this is probably a mistake.
-      let! results = Ply.List.mapSequentially (eval state st) exprs
+      let! results = exprs |> Ply.List.mapSequentially (eval state st)
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibExecution/Interpreter.fs:79]: Hope this isn't too nitpicky - I find it much easier to read relying on pipes.

File: fsharp-backend/src/LibExecution/Interpreter.fs [hunk 3]
```diff
@@ -409,7 +414,7 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
   }
 
 /// Interprets an expression and reduces to a Dark value
-/// (or task that should result in such)
+/// (or a task that results in a Dark value)
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibExecution/Interpreter.fs:417]:
```suggestion
/// (or a task that results in a dval)
```
Don't repeat "Dark value"

File: fsharp-backend/src/LibExecution/Interpreter.fs (hunk 1 above)
Comment (StachuDotNet) [reply fsharp-backend/src/LibExecution/Interpreter.fs]: Thanks!

File: fsharp-backend/src/LibExecution/Interpreter.fs (hunk 2 above)
Comment (pbiggar) [reply fsharp-backend/src/LibExecution/Interpreter.fs:85]: I would prefer we revert these changes. I'm not against pipeing but I find that it's more confusing when there's only one stage in the pipe.

File: fsharp-backend/src/LibExecution/Interpreter.fs [hunk 4]
```diff
@@ -203,17 +214,17 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
         // live values.
         let fakeST = Map.add "var" (DIncomplete SourceNone) st
         do! preview fakeST body
+
       let parameters =
-        List.choose
-          (function
+        parameters
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/Interpreter.fs:219]: I like this use of piping.

File: fsharp-backend/src/LibExecution/Interpreter.fs [hunk 5]
```diff
@@ -203,17 +214,17 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
...
+        |> List.choose (function
           | _, "" -> None
           | id, name -> Some(id, name))
-          parameters
 
       // It is the responsibility of wherever executes the DBlock to pass in
       // args and execute the body.
       return DFnVal(Lambda { symtable = st; parameters = parameters; body = body })
-    | EMatch (id, matchExpr, cases) ->
+    | EMatch (id, matchExpr, caseExprs) ->
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/Interpreter.fs:227]: I think "cases" is a better name here. The values in caseExprs are not expressions, they're a list of pattern+expr.

File: fsharp-backend/src/LibExecution/RuntimeTypes.fs [hunk 6]
```diff
@@ -153,20 +157,20 @@ type Expr =
...
-  /// - an expression resulting in a value that contains a field
-  /// - the name of the field to acceess
-  ///
-  /// <code>someExpr.fieldName</code>
+  /// Access a field of some expression (e.g. `someExpr.fieldName`)
   | EFieldAccess of id * Expr * string
 
+  /// Reference some local variable by name
+  ///
+  /// i.e. after a `let binding = value`, any use of `value`
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/RuntimeTypes.fs:165]:
```suggestion
  /// i.e. after a `let binding = value`, any use of `binding`
```

File: fsharp-backend/src/LibExecution/RuntimeTypes.fs [hunk 7]
```diff
@@ -363,7 +367,11 @@ and DType =
 /// information later, such as the iteration count that led to this, or
 /// something like a stack trace
 and DvalSource =
+  /// Function was either called directly,
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/RuntimeTypes.fs:370]: What does "function was called directly" mean? Why wouldn't this have a SourceID?

File: fsharp-backend/src/LibExecution/RuntimeTypes.fs [hunk 8]
```diff
@@ -841,8 +849,17 @@ and FnImpl =
...
   | PackageFunction of tlid * Expr
 
+
+// CLEANUP consider renaming to `ExecutionType`, `EvaluationMode`, etc.
+/// Represents the context in which we're evaluating some code
 and RealOrPreview =
+  /// We are evaluating an expression normally, and it is OK to affect state.
   | Real
+
+  /// We are previewing the evaluation of some expression,
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/RuntimeTypes.fs:859]: "within a canvas" isn't a precise definition here - all execution happens within a canvas. Perhaps something like "in the editor". Previews only happen in Wasm, we don't do it any other place.

File: fsharp-backend/src/LibExecution/Interpreter.fs [hunk 9]
```diff
@@ -33,10 +35,18 @@ let withGlobals (state : ExecutionState) (symtable : Symtable) : Symtable =
 let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
   // Design doc for execution results and previews:
   // https://www.notion.so/darklang/Live-Value-Branching-44ee705af61e416abed90917e34da48e
-  // TODO document is either gone or hidden behind login
+  // TODO remove link from code or avail document - it is either gone or hidden behind login
   let sourceID id = SourceID(state.tlid, id)
   let incomplete id = DIncomplete(SourceID(state.tlid, id))
 
+  /// This function ensures any value not on the exeuction path is evaluated.
```
[the remark this answers is not in the document]
Comment (StachuDotNet) [reply fsharp-backend/src/LibExecution/Interpreter.fs:42]: Thanks, resolving in general follow-up commit.

File: fsharp-backend/src/LibExecution/RuntimeTypes.fs (hunk 7 above)
Comment (StachuDotNet) [reply fsharp-backend/src/LibExecution/RuntimeTypes.fs:370]: I was making a bit of a guess here - removing that part of the comment.

**Pull request merged**